src/plumpy/base/state_machine.py

```diff
@@ -300,16 +314,23 @@ def _fire_state_event(self, hook: Hashable, state: Optional[State]) -> None:
     def on_terminated(self) -> None:
         """Called when a terminal state is entered"""
 
-    def transition_to(self, new_state: Union[Hashable, State, Type[State]], *args: Any, **kwargs: Any) -> None:
+    def transition_to(self, new_state: State | None, **kwargs: Any) -> None:
+        """Transite to the new state.
+
+        The new target state will be create lazily when the state is not yet instantiated,
+        which will happened for states not in the expect path such as pause and kill.
+        The arguments are passed to the state class to create state instance.
+        (process arg does not need to pass since it will always call with 'self' as process)
+        """
         assert not self._transitioning, 'Cannot call transition_to when already transitioning state'
 
+        if new_state is None:
+            return None
```

Comment:
> I think it makes more sense whoever is calling transition_to(None) is better not to call it.

This we can not ensure. The `transition_to` method can be called also for the terminal state. Otherwise `if new_state is None` should used every time before `transition_to` is called.